[Product Partnerships] Added SimpleWorkFlow Entity (#13856)
newrelic/docs-website: src/content/docs/infrastructure/amazon-integrations/aws-integrations-list/amazon-simpleworkflow-monitoring-integration.mdx

---
title: Amazon SimpleWorkFlow integration
tags:
  - Integrations
  - Amazon integrations
  - AWS integrations list
metaDescription: "New Relic's Amazon SimpleWorkFlow integration: what data it reports and how to enable it."
---

[New Relic infrastructure integrations](/docs/infrastructure/integrations-getting-started/getting-started/introduction-infrastructure-integrations) include an integration for AWS SimpleWorkFlow, sending its metrics data to New Relic.

This document explains the integration's features, how to activate it, and what data can be reported.

## Features

> Use our integration to collect and send telemetry data to New Relic from your [SimpleWorkFlow](https://aws.amazon.com/swf/). You can monitor your services, query incoming data, and build dashboards to observe everything at a glance.

Reviewer: ```suggestion
Use our integration to collect and send telemetry data to New Relic from your [Amazon Simple Workflow Service](https://aws.amazon.com/swf/). You can monitor your services, query incoming data, and build dashboards to observe everything at a glance.
```

## Activate integration

> To enable this integration, see how to [connect AWS services to New Relic](/docs/infrastructure/amazon-integrations/connect/aws-metric-stream-setup/).

Reviewer: ```suggestion
To enable this integration, see [how to connect AWS services to New Relic](/docs/infrastructure/amazon-integrations/connect/aws-metric-stream-setup/).
```

## Find and use data

> To find your integration's metrics, go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities)** > **Metrics and events** and filter by `aws.SimpleWorkFlow`.

Reviewer: ```suggestion
To find your integration's metrics, go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Metrics and events** and filter by `aws.SimpleWorkFlow`.
```

> ## Activate integration
> 
> To enable this integration, see [how to connect AWS services to New Relic](/docs/infrastructure/amazon-integrations/connect/aws-metric-stream-setup/).
> 
> ## Find and use data
> 
> To find your integration's metrics, go to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Metrics and events** and filter by `aws.SimpleWorkFlow`.

Reviewer: ```suggestion
## Get started

To enable this integration, see [how to connect AWS services to New Relic](/docs/infrastructure/amazon-integrations/connect/aws-metric-stream-setup/).

Once the integration is set up, you can find your metric data by going to **[one.newrelic.com](https://one.newrelic.com/all-capabilities) > Metrics and events** and filter by `aws.SimpleWorkFlow`.
```

Reviewer: @rahul188 What do you think about this change?

Author: @ally-sassman it seems good.

## Metric data [#metrics]

> This New Relic infrastructure integration collects the following Amazon SimpleWorkFlow data:

Reviewer: ```suggestion
This New Relic infrastructure integration collects the following Amazon SWF data:
```

> ### SimpleWorkFlow Metric data

Reviewer: ```suggestion
### SWF metric data
```

      </td>
    </tr>
    <tr>
      <td>
      `WorkflowsCompleted`
      </td>
      <td>
      Count
      </td>
      <td>
      The count of workflows that completed.
      </td>
    </tr>
    <tr>
      <td>
      `WorkflowsContinuedAsNew`
      </td>
      <td>
      Count
      </td>
      <td>
      The count of workflows that continued as new.
      </td>
    </tr>
    <tr>
      <td>
      `WorkflowsFailed`
      </td>
      <td>
      Count
      </td>
      <td>
      The count of workflows that failed.
      </td>
    </tr>
    <tr>
      <td>
      `WorkflowsTerminated`
      </td>
      <td>
      Count
      </td>
      <td>
      The count of workflows that were terminated.
      </td>
    </tr>
    <tr>
      <td>
      `WorkflowsTimedOut`
      </td>
      <td>
      Count
      </td>
      <td>
      The count of workflows that timed out, for any reason.
      </td>
    </tr>
    </tbody>
</table>

All imported data has one dimension: `Domain`

### Create alerts

You can set up alerts to notify you if there are any changes. For example, you can set up an alert to notify relevant parties of critical or fatal errors.

> Learn more about creating alerts [here](/docs/alerts-applied-intelligence/new-relic-alerts/get-started/your-first-nrql-condition/).

Reviewer: ```suggestion
Learn more about [creating alerts here](/docs/alerts-applied-intelligence/new-relic-alerts/get-started/your-first-nrql-condition/).
```
